simplq/src/main/java/me/simplq/service/TokenService.java

```diff
@@ -57,6 +58,11 @@ public TokenDetailResponse getToken(String queueId, String contactNumber) {
   @Transactional
   public TokenDeleteResponse deleteToken(String tokenId) {
     tokenRepository.setTokenStatusById(TokenStatus.REMOVED, tokenId);
+    /*Set Token Deletion timestamp to now*/
+    Date nowTime = new Date();
+
+    tokenRepository.findById(tokenId).get().setTokenDelete( nowTime);
+
```

Comment:
let us write it in a single statement:

```
// Set Token Deletion timestamp to now
tokenRepository.findById(tokenId).get().setTokenDelete(new Date());
```

Reply:
Also `findById` returns an Optional and if is empty `get()` will throw an Exception. You should use `map` to set token delete or else throw a business exception. Also if deleting a token is just add a timestamp, you can create a method `delete()` which will execute and update query where deletion is set to `NOW()`

Reply:
So the code will be much more clear 
```java
tokenRepository.findById(tokenId).map(token -> token.delete()).orElseThrow(...)
```

Reply:
Yeah, right. Let's create `delete()` as a separate refactor PR, for now:
```
var token = tokenRepository.findById(tokenId)
  .map(token -> {
    token.setTokenDelete(new Date());
    token.setTokenStatus(TokenStatus.REMOVED);
    tokenRepository.save(token);
    return new TokenDeleteResponse(
                    token.getId(), token.getQueue().getQueueName(), token.getStatus()));
  })
  .orElseThrow(SQInvalidRequestException::tokenNotFoundException);

```
(Not tested, might be missing a bracket)